### ISSUE TITLE
`UnusedVariable` false positive with guards

### ISSUE THREAD

**Issue by mgroth0** [detekt/detekt#7818]
* Version of detekt used: commit da8b164c5da3a2c8017452355c1754df7344c8a7

```kotlin
fun main() {
    val a = '1'.digitToInt() + 1
    val b = true // false positive UnusedVariable on `b`
    val c = false
    when (a) {
        1 if b -> Unit
        2      -> if (c) Unit else Unit
    }
}
```

**Comment by cortinico** [comment]
I suspect that is because of Kotlin 2.1

**Comment by atulgpt** [comment]
Hi @BraisGabin is guards api in kotlin final? If yes then this issue can be worked upon

**Comment by BraisGabin** [comment]
> Hi @BraisGabin is guards api in kotlin final? If yes then this issue can be worked upon

I have no idea. But we should support even if it is experimental. Our users should be free to use experimental features freely without detekt blocking them. So, if you want, you can go for it. 

This "rule" for sure has exceptions if the work vs benefit is huge then we probably would not support it.

**Comment by atulgpt** [comment]
@BraisGabin @cortinico after these rules migrated to Analysis API, these cases seems to be fixed. I have added TC at https://github.com/detekt/detekt/pull/8483

**Comment by BraisGabin** [comment]
Closing. The next release will have the fix. Thanks for the test cases @atulgpt :)
